commit 815314b86046bed5d3443c3af18c16f16401bc60 (refs/changes/04/3604/1)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2018-03-13T12:32:23+10:30 (1 year, 7 months ago)
    
    TRITON-196 cloudapi machine objects don't show "deletion_protection: false" if it isn't set

diff --git a/lib/machines.js b/lib/machines.js
index efdc628..85b3c7f 100644
--- a/lib/machines.js
+++ b/lib/machines.js
@@ -140,6 +140,7 @@ function translate(machine, req)  {
         ips: [],
         memory: Number(machine.ram),
         disk: (Number(machine.quota) * 1024) || 0,
+        deletion_protection: !!machine.indestructible_zoneroot,
         metadata: machine.customer_metadata || {},
         tags: machine.tags,
         credentials: credentials,
@@ -164,10 +165,6 @@ function translate(machine, req)  {
         msg.disk = Number(machine.disks[1].size);
     }
 
-    if (machine.indestructible_zoneroot) {
-        msg.deletion_protection = true;
-    }
-
     if (machine.nics && machine.nics.length) {
         machine.nics.forEach(function (nic) {
             msg.ips.push(nic.ip);
